### PR TITLE
feat(zip): hooks

### LINKED ISSUE #962: Add `zip:*` hooks
### Feature Request

Add the following zip hooks:

- `zip:start`
- `zip:extension:start`
- `zip:extension:done`
- `zip:sources:start`
- `zip:sources:done`
- `zip:done`

#### Is your feature request related to a bug?

N/A

### What are the alternatives?

Using a separate script.

### Additional context

I'm trying to write a module that outputs a self-signed XPI file using `web-ext`, and being able to do this in a module would be great.

### PR DESCRIPTION
**Pull request by Timeraa**
Closes #962

### CODE REVIEW

**Comment by Timeraa** [comment]
No idea why my branch also updates the gif but I guess it doesn't hurt

File: packages/wxt/src/core/zip.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import { normalizePath } from './utils/paths';
  */
 export async function zip(config?: InlineConfig): Promise<string[]> {
   await registerWxt('build', config);
+  await wxt.hooks.callHook('zip:start', wxt);
```
Comment (aklinker1) [inline packages/wxt/src/core/zip.ts:22]: Let's move `zip:start` to after we perform the build, right before we start zipping things. We already have other hooks like `build:start` that fire around this time.

File: packages/wxt/src/types.ts [hunk 2]
```diff
@@ -418,6 +418,45 @@ export interface WxtHooks {
    * @param viteConfig The config that will be used to build the entrypoints. Can be updated by reference.
    */
   'vite:devServer:extendConfig': (config: vite.InlineConfig) => HookResult;
+
+  /**
+   * Called before the zip process starts.
```
Comment (aklinker1) [inline packages/wxt/src/types.ts:423]:
```suggestion
   * Called after building the extension, right before WXT begins zipping files.
```

**Review by aklinker1 — CHANGES_REQUESTED** [review]
Nice, thanks, this looks good!

**Comment by aklinker1** [comment]
Checks are failing because you need to add the new hooks to the E2E test: https://github.com/wxt-dev/wxt/blob/main/packages/wxt/e2e/tests/hooks.test.ts

File: packages/wxt/src/core/zip.ts (hunk 1 above)
Comment (Timeraa) [reply packages/wxt/src/core/zip.ts:22]: True

**Comment by Timeraa** [comment]
> Checks are failing because you need to add the new hooks to the E2E test: https://github.com/wxt-dev/wxt/blob/main/packages/wxt/e2e/tests/hooks.test.ts

Opps forgot to run them because I am used to CI letting me know haha

**Comment by Timeraa** [comment]
I made the zip test add firefox as the browser so it also calls the sources hooks

**Review by aklinker1 — APPROVED** [review]

**Pull request merged**